### PR TITLE
Add autoprefix command

### PR DESCRIPTION
**Pull request by ImUrX**
This command lets you autoprefix in all your messages a text, it can be a command or a simple text.
It's intended use is for `/msg <user>` or `/tmsg` but it's a flexible addition to the mod :3

### CODE REVIEW

File: src/main/java/net/earthcomputer/clientcommands/ClientCommands.java
Comment (xpple) [inline src/main/java/net/earthcomputer/clientcommands/ClientCommands.java]: The `d` comes before the `t`, usually.

File: src/main/java/net/earthcomputer/clientcommands/command/AutoPrefixCommand.java
Comment (xpple) [inline src/main/java/net/earthcomputer/clientcommands/command/AutoPrefixCommand.java]: Please use wildcard imports.

File: src/main/java/net/earthcomputer/clientcommands/command/AutoPrefixCommand.java
Comment (xpple) [inline src/main/java/net/earthcomputer/clientcommands/command/AutoPrefixCommand.java]: I feel like this entire command can be a config instead, but I will continue reviewing the command anyway.

File: src/main/java/net/earthcomputer/clientcommands/command/AutoPrefixCommand.java
Comment (xpple) [inline src/main/java/net/earthcomputer/clientcommands/command/AutoPrefixCommand.java]: You're using a continuation indent of both 4 and 8, please use only 4.

File: src/main/java/net/earthcomputer/clientcommands/ClientCommands.java
Comment (Earthcomputer) [inline src/main/java/net/earthcomputer/clientcommands/ClientCommands.java]: This should be after `AuditMixinsCommand`

File: src/main/java/net/earthcomputer/clientcommands/command/AutoPrefixCommand.java
Comment (Earthcomputer) [inline src/main/java/net/earthcomputer/clientcommands/command/AutoPrefixCommand.java]: Could you make sure the indent is consistently 4 spaces?

File: src/main/java/net/earthcomputer/clientcommands/command/AutoPrefixCommand.java
Comment (Earthcomputer) [inline src/main/java/net/earthcomputer/clientcommands/command/AutoPrefixCommand.java]: This method and the field should be marked as `@Nullable`. Or, why not default it to `""`?

File: src/main/java/net/earthcomputer/clientcommands/command/AutoPrefixCommand.java
Comment (xpple) [inline src/main/java/net/earthcomputer/clientcommands/command/AutoPrefixCommand.java]: Idk if I'm a fan of hardcoding these in. It might get confusing and what if for some reason someone wanted the literal string `msg` as prefix?

File: .gitignore
Comment (Earthcomputer) [inline .gitignore]: Would prefer these weren't here either tbh

**Review by xpple — DISMISSED** [review]

File: src/main/java/net/earthcomputer/clientcommands/command/AutoPrefixCommand.java
Comment (Earthcomputer) [inline src/main/java/net/earthcomputer/clientcommands/command/AutoPrefixCommand.java]: As discussed on discord, `greedyString` here. It should definitely not be a text component, as that's not what you're typing in the text box.

**Review by Earthcomputer — CHANGES_REQUESTED** [review]

File: src/main/java/net/earthcomputer/clientcommands/command/AutoPrefixCommand.java
Comment (ImUrX) [reply src/main/java/net/earthcomputer/clientcommands/command/AutoPrefixCommand.java]: this is very nice syntactic sugar, it will make things less painful for searching a user >~>

Comment (Earthcomputer) [reply src/main/java/net/earthcomputer/clientcommands/command/AutoPrefixCommand.java]: Maybe have a literal `custom` and then the `greedyString` custom prefix argument

File: .gitignore
Comment (ImUrX) [reply .gitignore]: you are killing me

File: src/main/java/net/earthcomputer/clientcommands/command/AutoPrefixCommand.java
Comment (ImUrX) [reply src/main/java/net/earthcomputer/clientcommands/command/AutoPrefixCommand.java]: should it be a config? Should this be persistent?

Comment (Earthcomputer) [reply src/main/java/net/earthcomputer/clientcommands/command/AutoPrefixCommand.java]: configs can be temporary

Comment (xpple) [reply src/main/java/net/earthcomputer/clientcommands/command/AutoPrefixCommand.java]: Temporary configs exist. Although you would lose the autocompletion for players, unless you resort to a hack. I'll let @Earthcomputer decide.

File: src/main/java/net/earthcomputer/clientcommands/command/AutoPrefixCommand.java
Comment (ImUrX) [reply src/main/java/net/earthcomputer/clientcommands/command/AutoPrefixCommand.java]: done

File: src/main/java/net/earthcomputer/clientcommands/command/AutoPrefixCommand.java
Comment (ImUrX) [reply src/main/java/net/earthcomputer/clientcommands/command/AutoPrefixCommand.java]: ah, i understand, you meant making this a config setting qwq *cries*

File: src/main/resources/assets/clientcommands/lang/en_us.json
Comment (RealRTTV) [inline src/main/resources/assets/clientcommands/lang/en_us.json]: nit: "Prefixing \"%s\" in all your following messages" is more consise.

File: src/main/java/net/earthcomputer/clientcommands/command/AutoPrefixCommand.java
Comment (RealRTTV) [inline src/main/java/net/earthcomputer/clientcommands/command/AutoPrefixCommand.java]: a short-circuit branch to get the prefix would be nice

File: src/main/java/net/earthcomputer/clientcommands/command/AutoPrefixCommand.java
Comment (RealRTTV) [inline src/main/java/net/earthcomputer/clientcommands/command/AutoPrefixCommand.java]: personally, i'd prefer having root level `tmsg` and `msg` as pre-made prefixes, then `/cautoprefix set <value>` can be direct rather than requiring another depth of argument

your preference though

File: src/main/java/net/earthcomputer/clientcommands/command/AutoPrefixCommand.java
Comment (ImUrX) [reply src/main/java/net/earthcomputer/clientcommands/command/AutoPrefixCommand.java]: just query lazy

File: src/main/resources/assets/clientcommands/lang/en_us.json
Comment (ImUrX) [reply src/main/resources/assets/clientcommands/lang/en_us.json]: i prefer the current one, but maintainers are free to say other wise

File: src/main/java/net/earthcomputer/clientcommands/command/AutoPrefixCommand.java
Comment (ImUrX) [reply src/main/java/net/earthcomputer/clientcommands/command/AutoPrefixCommand.java]: will keep it as is then :p

File: src/main/resources/assets/clientcommands/lang/en_us.json
Comment (xpple) [reply src/main/resources/assets/clientcommands/lang/en_us.json]: I like the current one better.

**Comment by ImUrX** [comment]
Merge when? 🥺

**Comment by xpple** [comment]
Well there is an open question, so not yet.

File: src/main/java/net/earthcomputer/clientcommands/command/AutoPrefixCommand.java
Comment (RealRTTV) [reply src/main/java/net/earthcomputer/clientcommands/command/AutoPrefixCommand.java]: I second this, I think `null` and `""` should be their own separate things.

File: src/main/java/net/earthcomputer/clientcommands/command/AutoPrefixCommand.java
Comment (Earthcomputer) [reply src/main/java/net/earthcomputer/clientcommands/command/AutoPrefixCommand.java]: Yes please move it to a config

Comment (ImUrX) [reply src/main/java/net/earthcomputer/clientcommands/command/AutoPrefixCommand.java]: Done, haven't tested it tho

**Review by Earthcomputer — APPROVED** [review]

**Pull request merged**
